riscv/insns/vrol_vx.h

```diff
@@ -0,0 +1,21 @@
+// vrol.vx vd, vs2, rs1, vm
+
+#include "zvk_ext_macros.h"
+
+require_zvbb;
+
+// 'mask' selects the low log2(vsew) bits of the shift amount,
+// to limit the maximum shift to "vsew - 1" bits.
+const reg_t mask = P.VU.vsew - 1;
+
+// For .vx, the shift amount comes from rs1.
+const reg_t shift = ((reg_t)RS1) & mask;
+
+VI_VX_ULOOP
+({
+  if (shift != 0) {  // Protect against UB.
+    vd = (vs2 << shift) | (vs2 >> (sew - shift));
+  } else {
+    vd = vs2;
+  }
```

Comment:
I think this would be cleaner as `vd = (vs2 << shift) | (vs2 >> (-shift & (sew - 1)));`, which fixes the UB in the same way as we did for the scalar `rol` instruction.

Reply:
Done, thanks for the suggestion.

I did a bit more work around the rotate logic in those commits:
 - I introduced rotate_right/rotate_left template functions in arith.h . 
 - I use those functions in the ROR/ROL macros in zvk_ext_macros.h .
 - I added "ZVK_" prefixes to the ROR/ROL macros to reduce the risk of collisions

The vector instructions still use their own logic rather than use the rotate_right/rotate_left versions. Let me know if you think we should change that.